src/Dropbox.elm

```diff
@@ -861,7 +924,148 @@ upload auth info =
             ]
         , url = url
         , body = body
-        , expect = Http.expectJson decodeUploadResponse
+        , expect = Http.expectJson decodeFileMetadata
+        , timeout = Nothing
+        , withCredentials = False
+        }
+        |> Http.toTask
+        |> Task.mapError decodeError
+
+
+{-| Request parameters for `listFolder`
+-}
+type alias ListFolderRequest =
+    { path : String
+    , recursive : Bool
+    , includeMediaInfo : Bool
+    , includeDeleted : Bool
+    , includeHasExplicitSharedMembers : Bool
+    }
+
+
+{-| See <https://www.dropbox.com/developers/documentation/http/documentation#files-list_folder>
+-}
+type ListFolderError
+    = PathListError LookupError
+    | ExpiredCursorError
```

Comment:
That's from the [list_folder/continue](https://www.dropbox.com/developers/documentation/http/documentation#files-list_folder-continue) API. On reflection, I think having a separate `ListFolderError` and `ListFolderContinueError` makes more sense than sharing a union type which is partially inapplicable to one API call.  It also fits better with your point about making the code similar to what an auto-generated API would look like.